Reduce extra output in compilation and tests

hubertp opened the pull request:
### Pull Request Description

I couldn't stand the amount of extra output that we got when compiling a clean project and when executing regular tests. We should strive to keep output clean and not print anything additional to stdout/stderr.


### Important Notes

No feature changes. This is a purely setup issue. Should see much less clutter when building/running tests.
To avoid explicit setup within the code, we provide a `TestLogProvider`. But since we also use logback both conflict when slf4j searches for providers. To avoid warnings (and potential problems if it decides to pick up the logback one) we explicitly say which one should be selected. Support for `-Dslf4j.provider` has been only added in 2.x, hence the upgrade.

### Checklist
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.

hubertp committed:
Reduce extra output in compilation and tests

I couldn't stand the amount of extra output that we got when compiling
a clean project and when executing regular tests. We should strive to
keep output clean and not print anything additional to stdout/stderr.

JaroslavTulach reviewed (commented):
Suppressing output when the test succeeds is certainly fine. However not displaying any logged messages when the test fail isn't. I share your preference of not polluting the output unnecessarily, but the current change (as far as I understand it) will just make understanding of (transient) test failures harder.

Can we get [NetBeans Test Case logLevel()](https://github.com/apache/netbeans/blob/f75da6ef22b2c637ee50ab10cc53a695a7c47a1e/harness/nbjunit/src/org/netbeans/junit/NbTestCase.java#L264) support for our Scala tests?

JaroslavTulach commented on engine/runtime-with-instruments/src/test/scala/org/enso/interpreter/test/instrument/RuntimeProjectContextTest.scala:
```diff
@@ -12,6 +13,7 @@ class RuntimeProjectContextTest extends AnyWordSpec with Matchers {
     "report an exception if ran in context of a project " +
     "which cannot be loaded" in {
       val thrown = intercept[PolyglotException] {
+        LoggerSetup.get().setup()
```
The necessity to call this `LoggerSetup.get().setup()` is annoying as also [noted elsewhere](https://github.com/enso-org/enso/pull/7803/files#r1326850557). The rule for these [injectable singletons](http://wiki.apidesign.org/wiki/Injectable_Singleton) is that they should be _configured automatically_ otherwise they become [pathological liars](http://wiki.apidesign.org/wiki/Singleton#Singletons_are_Pathological_Liars.5B3.5D).

Looks like it is not the case and that's wrong. We should find a way to convince whatever logging framework we use to pick whatever configuration `LoggerSetup.get().setup()` does up without any initialization call from tests. May we provide [our own provider](https://www.slf4j.org/manual.html#swapping) to make sure a call to `LoggerSetup.get().setup()` is made?

hubertp replied:
Provider was added, thanks for the suggestion!

Akirathan commented on engine/runtime/src/test/java/org/enso/compiler/SerdeCompilerTest.java:
```diff
@@ -142,7 +142,7 @@ public void publish(LogRecord lr) {
         failure = new AssertionError(this.toString() + "\nGot forbidden message: " + msg);
       }
-      System.err.println(msg);
+      // System.err.println(msg);
```
Remove commented code?

Akirathan reviewed (approved):
Very nice DevX improvement.

hubertp committed:
Getting rid of explicit setup by service loading

In order for SL4J to use service loading correctly had to upgrade to
latest slf4j. Unfortunately `TestLogProvider` which essentially
delegates to `logback` provider will lead to spurious ambiguous warnings
on multiple providers. In order to dictate which one to use and
therefore eliminate the warnings we can use the `slf4j.provider` env
var, which is only available in slf4j 2.x.

Now, there is no need to explicitly call `LoggerSetup.get().setup()` as
that is being called during service setup.

hubertp committed:
legal review

hubertp committed:
linter

hubertp committed:
Ensure ConsoleHandler uses the default level

ConsoleHandler's constructor uses `Level.INFO` which is unnecessary for
tests.

hubertp committed:
report warnings
